Add Default Item Quality Value option into Options

mateokladaric opened the pull request:
Fixes #1569  .

### Description of the problem being solved:
Forcing 20 quality on items crafted or imported without an option to change the default value assigned.

### Steps taken to verify a working solution:
- Crafted and imported an item from PoE2

### Link to a build that showcases this PR:
--- Any build just craft or ctrl v an item into items tab.

### Before screenshot:
<img width="414" height="219" alt="image" src="https://github.com/user-attachments/assets/eb9e0e23-f825-4f62-8a7d-2f746e6d069b" />

### After screenshot:
<img width="333" height="84" alt="image" src="https://github.com/user-attachments/assets/55f7da95-7a7e-48ae-bffc-cd1666cbe45a" />
<img width="390" height="205" alt="image" src="https://github.com/user-attachments/assets/7635158a-e047-4629-b8e5-64407dc5f4d8" />

mateokladaric committed:
add default item quality option in settings

LocalIdentity commented on src/Modules/Main.lua:
```diff
@@ -102,6 +102,7 @@ function main:Init()
 	self.thousandsSeparator = ","
 	self.decimalSeparator = "."
 	self.defaultItemAffixQuality = 0.5
+	self.defaultItemQuality = 0
```
Should default to 20 quality

LocalIdentity commented on src/Modules/Main.lua:
```diff
@@ -782,7 +786,8 @@ function main:SaveSettings()
 		betaTest = tostring(self.betaTest),
 		edgeSearchHighlight = tostring(self.edgeSearchHighlight),
-		defaultGemQuality = tostring(self.defaultGemQuality or 0),
+			defaultGemQuality = tostring(self.defaultGemQuality or 0),
```
Looks like indenting is off by 1 tab length

mateokladaric committed:
speedy changes

LocalIdentity committed:
Set max default to 30

Commit:
Fix save issue

The value was not reverting back to the original value on pressing cancel
Also missing 1 other default to 20 check
